### DISCUSSION TITLE
Check if dpoint is included inside polygon.

### DISCUSSION THREAD

**Discussion by pfeatherstone** [davisking/dlib#2859] [Q&A] [answered]
Is there a way to check a point is inside a polygon?
Context: I'm using mediapipe's facemesh model to extract 478 face landmarks. Some of them give you contours around the face, eyes, eyebrows and mouth. I want to get all pixels on the face that map to skin. I think I need 6 polygons: face, left eye, right eye, left eyebrow, right eyebrow and mouth. Then for all pixels in the face crop, pick those which are inside the face contour but not inside the others. Make sense?
But I need some code that checks whether a point is inside some arbitrary convex polygon. Is there a dlib function for this? Cheers. Here is an example face, with landmark detections and contours.
![mediapipe](https://github.com/davisking/dlib/assets/45853521/134087b2-8e60-42c1-a520-fa829ffccff6)

**Comment by pfeatherstone** [comment]
Basically, is there an equivalent of `cv::pointPolygonTest()` ? @arrufat You know dlib like the back of your hand.

**Reply by arrufat** [reply]
As far as I know there isn't.

I had to do something similar recently and my approach was to generate a mask by:
- painting the face with white
- painting the other parts with black

All that using draw_solid_polygon.

Then I just use that mask to check if the pixel is white or not...

I'm open to faster/more efficient ways.

**Reply by pfeatherstone** [reply]
I think this is what i want cheers

**Reply by arrufat** [reply]
I am fairly new to GitHub discussions. Could you try marking this as "Answered" so that people can filter by answered discussions?

**Comment by arrufat** [comment, marked as answer]
I put the answer as a comment, that's why it didn't count as an answer, and it looks like the question remains unanswered.

Here's my answer (copied from my [comment](https://github.com/davisking/dlib/discussions/2859#discussioncomment-6991235) above) so that we can mark this discussion as "answered":


> As far as I know there isn't.
> 
> I had to do something similar recently and my approach was to generate a mask by:
> 
>   - painting the face with white
>   - painting the other parts with black
> 
> All that using draw_solid_polygon.
> 
> Then I just use that mask to check if the pixel is white or not...
> 
> I'm open to faster/more efficient ways.
